xbmc/utils/SeekHandler.cpp

```diff
@@ -130,56 +128,44 @@ void CSeekHandler::Seek(bool forward, float amount, float duration /* = 0 */, bo
   // not yet seeking
   if (!m_requireSeek)
   {
-    m_percent = static_cast<float>(g_infoManager.GetPlayTime()) / g_infoManager.GetTotalPlayTime() * 0.1f;
-    m_percentPlayTime = m_percent;
-
     // tell info manager that we have started a seek operation
     m_requireSeek = true;
     g_infoManager.SetSeeking(true);
     m_seekStep = 0;
+    m_seekSize = 0;
     m_analogSeek = analogSeek;
     m_seekDelay = analogSeek ? analogSeekDelay : m_seekDelays.at(type);
   }
 
   // calculate our seek amount
-  if (!g_infoManager.m_performingSeek)
+  if (analogSeek)
   {
-    if (analogSeek)
-    {
-      //100% over 1 second.
-      float speed = 100.0f;
-      if( duration )
-        speed *= duration;
-      else
-        speed /= g_infoManager.GetFPS();
+    //100% over 1 second.
+    float speed = 100.0f;
+    if( duration )
+      speed *= duration;
+    else
+      speed /= g_infoManager.GetFPS();
 
-      if (forward)
-        m_percent += amount * amount * speed;
-      else
-        m_percent -= amount * amount * speed;
+    int seekSize = (amount * amount * speed) * g_infoManager.GetTotalPlayTime() / 100;
```

Comment:
how often do you want to touch it? it is still broken for material having no total playing time, hence all those changes are fixes not improvements. I see no reason why we should continue to do it wrong. let's eliminate the flaws which are the dependency to infomanager and lack of relative seek capability of player.